test/controllers/publishers/site_banners_controller_test.rb

```diff
@@ -23,4 +23,48 @@ class PublishersControllerTest < ActionDispatch::IntegrationTest
     publisher.reload
     assert_equal publisher.site_banner, SiteBanner.last
   end
+
+  test "publisher cannot upload an excessively large file" do
+    publisher = publishers(:completed)
+    sign_in publisher
+
+    get home_publishers_path
+    assert_response :success
+
+    post(publisher_site_banners_path(publisher),
+      params: {
+        title: "Hello World",
+        description: "Lorem Ipsum",
+        donation_amounts: "[5, 10, 15]",
+      }
+    )
```

Comment:
Consider removing this json , since we're only testing file upload

```
params: {
        title: "Hello World",
        description: "Lorem Ipsum",
        donation_amounts: "[5, 10, 15]",
      }
```